scipy/stats/_stats_py.py

```diff
@@ -1517,9 +1517,11 @@ def describe(a, axis=0, ddof=1, bias=True, nan_policy='propagate'):
     xp = array_namespace(a)
     a, axis = _chk_asarray(a, axis, xp=xp)
 
-    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+    contains_nan = _contains_nan(a, nan_policy)
 
-    if contains_nan and nan_policy == 'omit':
+    # Test nan_policy before the implicit call to bool(contains_nan)
+    # to avoid raising on lazy xps on the default nan_policy='propagate'
```

Comment:
I'll iterate on this to print a nice error message if `is_lazy_array(contains_nan)` after https://github.com/data-apis/array-api-compat/issues/225 gets merged and published